### PR TITLE
Invalid dates

### PR DESCRIPTION
**Pull request by pwentz**
Add Ajax call to disable user from choosing nights that are already booked.

### CODE REVIEW

**Comment by Automatic365** [comment]
This feature is great. If I understand this correctly, then we don't need the validations?

File: app/models/property.rb [hunk 1]
```diff
@@ -14,4 +14,12 @@ class Property < ApplicationRecord
...
   has_many :booked_dates
   has_many :nights, through: :booked_dates
+
+  def bookings
+    reservations.map do |r|
+      Night.where(
+        date: r.starting_date...r.end_date
+      ).map{ |d| d.date.strftime('%-m/%-d/%Y') }
+    end.flatten
+  end
```
Comment (dtinianow) [inline app/models/property.rb]: This could be aligned better

**Comment by dtinianow** [comment]
This looks really good Pat.  Nice work

File: app/assets/javascripts/datepicker.js [hunk 2]
```diff
@@ -1,17 +1,43 @@
...
+      beforeShowDay: checkAvailability,
+      minDate: 0,
       onSelect: function(dateText, inst) {
-        console.log(dateText);
       }
     })
   });
 
+  let prop_id = window.location.pathname.slice(-1)
+  let bookings
```
Comment (Yoyo2Code) [inline app/assets/javascripts/datepicker.js]: Missing ";" at end of "bookings"

**Comment by pwentz** [comment]
Correct. With the invalid dates being disabled, the user cannot send any sort of POST request to the cart items controller with invalid information. This way we stop the problem a step before it reaches the database.

**Comment by Automatic365** [comment]
> Excellent

File: app/models/property.rb (hunk 1 above)
Comment (pwentz) [inline app/models/property.rb]: Yeah, I agree it did get sort of messy towards the end there. Will refactor.

**Comment by Yoyo2Code** [comment]
Awesome job on the AJAX calls and ES5, but there's missing semi-colons at the end of variables.

**Pull request merged**